katdal/concatdata.py

```diff
@@ -444,10 +442,31 @@ def __setitem__(self, name, data):
             for n, cache in enumerate(self.caches):
                 cache[name] = data[self._segments[n]:self._segments[n + 1]]
 
-    def iterkeys(self):
+    def __delitem__(self, name):
+        found = False
+        for cache in self.caches:
+            try:
+                del cache[name]
+                found = True
+            except KeyError:
+                pass
+        if not found:
+            raise KeyError
+
+    def __contains__(self, name):
+        return any(name in cache for cache in self.caches)
+
+    def __len__(self):
+        return sum(1 for _ in self)
```

Comment:
Nice trick to remember for iterators.

Reply:
I can't claim credit - I found it somewhere on the internet.